### ISSUE TITLE
Tyvar the Bellicose ability not resolving for all Elves

### ISSUE THREAD

**Issue by skaspels** [magefree/mage#11560]
I've been playing a Tyvar edh deck for some time, and when elves have a mana ability granted from another source, the mana ability doesn't trigger Tyvar when it resolves.

Ex: I have a Level 5 Joraga Treespeaker, Tyvar the Bellicose, and Gnarlroot Trapper. If I use Gnarlroot trapper's t: add {B} ability, Tyvar's ability triggers and Gnarlroot gets a +1/+1 counter. If instead I use Gnarlroot trapper's t: Add {GG} ability granted by Joraga Treespeaker, Tyvar's ability does not trigger. I have found this happens very consistently across games.

**Comment by xenohedron** [comment]
[[Tyvar, the Bellicose]]

Relevant code is here in the custom triggered ability, not immediately obvious to me what the issue is, so maybe buried in the `getAbility`:
```java
    public boolean checkTrigger(GameEvent event, Game game) {
        if (isControlledBy(event.getPlayerId())
                && event.getSourceId().equals(getSourceId())
                && game
                .getAbility(event.getTargetId(), event.getSourceId())
                .map(ManaAbility.class::isInstance)
                .orElse(false)) {
            this.getEffects().setValue("damage", event.getAmount());
            return true;
        }
        return false;
    }
```

**Comment by skaspels** [comment]
I would guess, based on what I'm reading there, that the problem is the 
`
                && event.getSourceId().equals(getSourceId())
 `               
part of the 'if' statement - the event's target (the creature) and the source (the thing giving the creature the mana ability) aren't the same. perhaps this needs to be modified.

**Comment by xenohedron** [comment]
Nope, looks like the problem is that `game.getAbility(event.getTargetId(), event.getSourceId())` is returning an empty `Optional` when it is an ability being granted to the creature from another source.

**Comment by xenohedron** [comment]
You know what, this might be related to #11077, if perhaps the gained ability has a new id by the time checkTrigger tries to find it.

**Comment by skaspels** [comment]
I guess that makes sense - the game is trying to distinguish between being the creature tapping for mana as a mana ability, vs being tapped by another card, like [[Relic of Legends]] (which wouldn't trigger Tyvar).

As to your second comment, I don't know what the ability would have a new ID when checkTrigger tries to find it, but then I don't understand the code very well at all.

**Comment by jeffwadsworth** [comment]
[[Joraga Treespeaker]] [[Gnarlroot Trapper]]

**Comment by jeffwadsworth** [comment]
The event MANA_ADDED fires whether a mana source is used or not.  Make sure to test cards like Gnarlroot Trapper before anything else.  Caged Sun is the only other card using this event trigger.
